### ISSUE TITLE
Cast issue when building from source

### ISSUE THREAD

**Issue by nicolaje** [ANYbotics/variant#6]
Building the packages from sources on Ubuntu 18.04, GCC 7.3.0 there are many error messages that prevent building this package:

```bash
variant_topic_tools/src/DataType.cpp:100:66: error: cannot convert ‘boost::shared_ptr<variant_topic_tools::ArrayDataType::Impl>’ to ‘bool’ in return
```

**Comment by andre-nguyen** [comment]
Anyone found a fix?

**Comment by andre-nguyen** [comment]
I've sent in a pr to fix this problem here #7 let me know if it also works for you.

**Comment by samuelba** [comment]
thanks for the pull request
